cupy/statistics/histogram.py

```diff
@@ -3,7 +3,70 @@
 import cupy
 
 
-# TODO(okuta): Implement histogram
+def histogram(x, bins=10):
+    """Computes the histogram of a set of data.
+
+    Args:
+        x (cupy.ndarray): Input array.
+        bins (int or cupy.ndarray): If ``bins`` is an int, it represents the
+            number of bins. If ``bins`` is an :class:`~cupy.ndarray`, it
+            represents a bin edges.
+
+    Returns:
+        tuple: ``(hist, bin_edges)`` where ``hist`` is a :class:`cupy.ndarray`
+        storing the values of the histogram, and ``bin_edges`` is a
+        :class:`cupy.ndarray` storing the bin edges.
+
+    .. seealso:: :func:`numpy.histogram`
+    """
+
+    if x.dtype.kind == 'c':
+        # TODO(unno): comparison between complex numbers is not implemented
+        raise NotImplementedError('complex number is not supported')
+
+    if isinstance(bins, int):
+        if x.size == 0:
+            min_value = 0.0
+            max_value = 1.0
+        else:
+            min_value = float(x.min())
+            max_value = float(x.max())
+        if min_value == max_value:
+            min_value -= 0.5
+            max_value += 0.5
+        bins = cupy.linspace(min_value, max_value, bins + 1)
+    elif isinstance(bins, cupy.ndarray):
+        if cupy.any(bins[:-1] > bins[1:]):
```

Comment:
Please use `>=`. See also [here](https://github.com/numpy/numpy/blob/v1.13.0/numpy/lib/function_base.py#L786).

Reply:
I referred to this code: https://github.com/numpy/numpy/blob/master/numpy/lib/histograms.py#L315

Reply:
Yes, you are right. But, old NumPy is a different behavior.

Reply:
If its behavior changed, we need to skip test with old NumPys.

Reply:
OK, Please fix test?

Reply:
I found that this behavior is a bug which occurs with unsigned integer in old numpy and is fixed in numpy 1.13.2.